custom_components/scout_alarm/sensor.py

```diff
@@ -0,0 +1,145 @@
+import json
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import HomeAssistantType
+import homeassistant.components.sensor as sensor
+
+from homeassistant.const import (
+    ATTR_ATTRIBUTION,
+    PERCENTAGE,
+    TEMP_CELSIUS
+)
+
+from homeassistant.components.sensor import (
+    DEVICE_CLASS_TEMPERATURE,
+    DEVICE_CLASS_HUMIDITY
+)
+
+from .const import (
+    ATTRIBUTION,
+    DOMAIN,
+    LOGGER,
+    SCOUT_DEVICE_TYPE_DOOR_PANEL,
+    SCOUT_DEVICE_TYPE_ACCESS_SENSOR,
+    SCOUT_DEVICE_TYPE_MOTION_SENSOR,
+    SCOUT_DEVICE_TYPE_WATER_SENSOR
+)
+
+SUPPORTED_SCOUT_DEVICE_TYPES_TEMPERATURE = [
+    SCOUT_DEVICE_TYPE_DOOR_PANEL,
+    SCOUT_DEVICE_TYPE_ACCESS_SENSOR,
+    SCOUT_DEVICE_TYPE_MOTION_SENSOR,
+    SCOUT_DEVICE_TYPE_WATER_SENSOR
+]
+
+SUPPORTED_SCOUT_DEVICE_TYPES_HUMIDITY = [
+    SCOUT_DEVICE_TYPE_DOOR_PANEL,
+    SCOUT_DEVICE_TYPE_ACCESS_SENSOR,
+    SCOUT_DEVICE_TYPE_MOTION_SENSOR
+]
+
+SENSOR_TYPES = {
+    "temperature": [TEMP_CELSIUS, DEVICE_CLASS_TEMPERATURE, "-T", "(T)"],
+    "humidity": [PERCENTAGE, DEVICE_CLASS_HUMIDITY, "-H", "(H)"],
+}
+
+async def async_setup_entry(hass: HomeAssistantType, config_entry: ConfigEntry, async_add_entities):
+    """Set up entry."""
+    entities = []
+    scout_alarm = hass.data[DOMAIN]
+    location_api = scout_alarm.location_api
+
+    devices = await location_api.get_devices()
+
+    for d in devices:
+        if d['type'] in SUPPORTED_SCOUT_DEVICE_TYPES_TEMPERATURE:
```

Comment:
Instead of basing this logic on the device type, can we do it by the presence of a `temperature` attribute under `reported` attribute in the device payload? e.g. 
```python
if d['reported'].get('temperature'):
```

Similar for `humidity`. My door panels are an older model and actually don't support temp or humidity readings.